ui/StatusQ/src/stringutilsinternal.cpp

```diff
@@ -32,27 +33,24 @@ QString StringUtilsInternal::escapeHtml(const QString& unsafe) const
     return unsafe.toHtmlEscaped();
 }
 
-QString StringUtilsInternal::readTextFile(const QString& filePath) const
+QString StringUtilsInternal::readTextFile(const QString& filePath)
 {
     auto engine = qmlEngine(this);
 
-    auto selector = QQmlFileSelector::get(engine);
+    QScopedPointer<QQmlFileSelector> selector(new QQmlFileSelector(engine, this));
```

Comment:
It doesn't seem correct.

According to docs:

> Creates a new QQmlFileSelector with parent object parent, which includes its own [QFileSelector](https://doc.qt.io/qt-6/qfileselector.html). engine is the [QQmlEngine](https://doc.qt.io/qt-6/qqmlengine.html) you wish to apply file selectors to. It will also take ownership of the QQmlFileSelector.

IIUC this way you can apply selector but not fetch the currently set ones. Moreover the engine is intended to take ownership, so removing the object at the end of scope is definitely wrong.

Reply:
Well I think it's the correct way to use it; it gets initialized with the (existing) engine as a parent. But anyway, I don't think we need the selector here at all; it was introduced in the past to read bundled JS files for the WalletConnect SDK... our current use cases are much simpler than this.